validator/validator-main.protoascii

```diff
@@ -4876,7 +4876,7 @@ tags: {
   }
   attrs: {
     name: "src"
-    value_regex: "data:image\\/svg\\+xml;charset=utf-8,<svg height=\"\\d+(\\.\\d+)?\" width=\"\\d+(\\.\\d+)?\" xmlns=\"http:\\/\\/www\\.w3\\.org\\/2000\\/svg\" version=\"1\\.1\"\\/>|data:image\\/svg\\+xml;charset=utf-8,<svg height='\\d+(\\.\\d+)?\' width='\\d+(\\.\\d+)?\' xmlns='http:\\/\\/www\\.w3\\.org\\/2000\\/svg' version='1\\.1'\\/>"
+    value_regex: "data:image\\/svg\\+xml;charset=utf-8,<svg height=\"\\d+(\\.\\d+)?\" width=\"\\d+(\\.\\d+)?\" xmlns=\"http:\\/\\/www\\.w3\\.org\\/2000\\/svg\" version=\"1\\.1\"\\/>|data:image\\/svg\\+xml;charset=utf-8,<svg height='\\d+(\\.\\d+)?\' width='\\d+(\\.\\d+)?\' xmlns='http:\\/\\/www\\.w3\\.org\\/2000\\/svg' version='1\\.1'\\/>|data:image\\/svg\\+xml;base64,[a-zA-Z0-9+\\/=]+"
```

Comment:
Inline SVGs have a different context than SVGs included by src. Inline SVGs can, for example, load another image over the network. They can also include a `<script>` tag which will execute in the document context.

This is not OK. If you have a very-specific literal you want to allow as an inline data svg, that's potentially OK. We can't allow any SVG bytes however.

Reply:
Is it feasible for the Validator to be told with some flag that the value is base64-encoded and should be decoded before matching the `value_regex`?

Reply:
> They can also include a `<script>` tag which will execute in the document context.

Actually, SVG images loaded in an `img` element do not execute scripts. 

That's a good thing because `amp-img` already allows for just plain `data:` protocol without any constraints on the image being loaded. For example, this `svg` image in a `data:` protocol is valid AMP even though it contains a `script`, but the script is not executed:

```html
<amp-img 
  src="data:image/svg+xml;charset=utf-8,<svg height='100' width='100' xmlns='http://www.w3.org/2000/svg' version='1.1'><circle cx='50' cy='50' r='50' /><script>console.info('Hello world!');</script></svg>" 
  width=100
  height=100
></amp-img>
```

Here's a test case to show this: https://amp-svg-script-testing.glitch.me/

Reply:
It's true that inline SVGs are bad, but in this context inline SVG is only SVG-tag-in-HTML-tree, not data: inlined SVGs. Data URL images shouldn't behave any differently than non-data-URL images.

Reply:
Thanks for the clarification @molnarg, I was indeed confused which side this case would fall into. I'll take another look at the PR today.